sample_apps/rust/examples/cleanup_sample.rs

```diff
@@ -0,0 +1,114 @@
+pub mod utils;
+use crate::utils::timestream_helper;
+use clap::Parser;
+use std::error::Error;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let args = timestream_helper::Args::parse();
+
+    let client = timestream_helper::get_connection(&args.region)
+        .await
+        .expect("Failed to get connection to Timestream");
+
+    match client
+        .describe_table()
+        .database_name(&args.database_name)
+        .table_name(&args.table_name)
+        .send()
+        .await
+    {
+        Ok(describe_table_output) => {
+            if let Some(table) = describe_table_output.table() {
+                if let Some(magnetic_store) = table.magnetic_store_write_properties() {
+                    if magnetic_store.enable_magnetic_store_writes() {
+                        if let Some(rejected_data_location) =
+                            magnetic_store.magnetic_store_rejected_data_location()
+                        {
+                            if let Some(s3_config) = rejected_data_location.s3_configuration() {
+                                let bucket_name = s3_config
+                                    .bucket_name()
+                                    .expect("Failed to retrieve bucket name");
+                                match timestream_helper::delete_s3_bucket(bucket_name, &args.region)
+                                    .await
+                                {
+                                    Ok(_) => {
+                                        println!("s3 bucket {:?} successfully deleted", bucket_name)
+                                    }
+                                    Err(err) => println!(
+                                        "Failed to delete s3 bucket {:?}, err: {:?}",
+                                        bucket_name, err
+                                    ),
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
You can greatly reduce this if-let nesting by pulling this logic out into a different function that returns `Option<()>` and using `?` on the options. Sort of like this:
```
let table = describe_table_output.table()?;
let magnetic_store = table.magnetic_store_write_properties()?;
if magnetic_store.enable_magnetic_store_writes() {
    let rejected_data_location = magnetic_store_rejected_data_location()?;
    let s3_config = rejected_data_location.s3_configuration()?;
    ...
}
```